File: content/en/04-llms/llms/_solutions/07_models/07_models.py
# %%
from chatlas import ChatLMStudio, ChatPosit

# %% [markdown]
# List models by calling the `list_models` method on a `Chat` instance.

# %%
ChatPosit().list_models()

# %% [markdown]
# You can also load the models into a Polars DataFrame for easier viewing.

# %%
import polars as pl

models = ChatPosit().list_models()
models = pl.DataFrame(models)
models

# %% [markdown]
# Now try sending the same prompt to different models to compare the responses.

# %%
prompt = "Write a recipe for an easy weeknight dinner."

ChatPosit().chat(prompt)
ChatPosit(model="zai-org/GLM-5.3").chat(prompt)

# %%
# If you have local models installed, try them out with LM Studio.
# (Local models only work on your own computer -- you can't use them on
# Posit Cloud. This line errors if LM Studio isn't running, so we've
# commented it out -- uncomment it to try it.)
# ChatLMStudio(model="prism-ml/bonsai-27b").chat(prompt)

File: content/en/04-llms/llms/_solutions/07_models/07_models.R
library(ellmer)

# List models using the `models_posit()` function.
# Hint: try using the Positron data viewer by calling `View()` on the results.
models_posit()

prompt <- "Write a recipe for an easy weeknight dinner."

# Try sending the same prompt to different models to compare the responses.
chat_posit()$chat(prompt)
chat_posit(model = "zai-org/GLM-5.3")$chat(prompt)

# If you have local models installed, you can use them too.
# (Local models only work on your own computer -- you can't use them on
# Posit Cloud. This line errors if LM Studio isn't running, so we've
# commented it out -- uncomment it to try it.)
# chat_lmstudio(model = "prism-ml/bonsai-27b")$chat(prompt)
